txpipe/extensions/twopoint_scia.py

```diff
@@ -0,0 +1,493 @@
+        'redshift_shearcatalog': False,
+        }
+
+    def run(self):
+
+        super().run()
+
+    def select_calculations(self, data):
+        source_list = data['source_list']
+        calcs = []
+        
+        if self.config['do_shear_pos']:
+            k = SHEAR_POS
+            l = SHEAR_POS_SELECT
+            for i in source_list:
+                calcs.append((i,i,k))
+                calcs.append((i,i,l))
+        
+        if self.config['do_pos_pos']:
+            if not 'random_bin' in data:
+                raise ValueError('You need to have a random catalog to calculate position-position correlations')
+            k = POS_POS
+            for i in source_list:
+                calcs.append((i,i,k))
+        
+        if self.config['do_shear_shear']:
+            k = SHEAR_SHEAR
+            for i in source_list:
+                for j in range(i+1):
+                    if j in source_list:
+                        calcs.append((i,j,k))
+
+
+        if self.rank==0:
+            print(f"Running these calculations: {calcs}")
+        
+        return calcs
+
+    def load_shear_catalog(self, data):
+
+        # Columns we need from the shear catalog
+        read_shear_catalog_type(self)
+
+        if self.config['shear_catalog_type']=='metacal':
+            if self.config['use_true_shear']:
+                cat_cols = ['ra', 'dec', 'true_g1', 'true_g2', 'mcal_flags']
+            else:
+                cat_cols = ['ra', 'dec', 'mcal_g1', 'mcal_g2', 'mcal_flags']
+                
+        else:
+            cat_cols = ['ra', 'dec', 'g1', 'g2', 'weight','flags','sigma_e','m']
+        print(f"Loading shear catalog columns: {cat_cols}")
+
+        f = self.open_input('shear_catalog')
+        g = f['shear']
+        for col in cat_cols:
+            print(f"Loading {col}")
+            data[col] = g[col][:]
+
+        if self.config['3Dcoords']:
+            #Temporary fix for not running PDF's on DES
+            if self.config['redshift_shearcatalog']:
+                data['mu'] = g['mean_z']
+            else: 
+                h = self.open_input('photoz_pdfs')
+                g = h['point_estimates']
+                data['mu'] = g['z_mean'][:]
+
+    def load_random_catalog(self, data):
+        # For now we are just bypassing this, since it is not needed
+
+        filename = self.get_input('random_cats_source')
+        if filename =='None':
+            filename = None
+
+        if filename is None:
+            print("Not using randoms")
+            return
+
+        
+        # Columns we need from the tomography catalog
+        randoms_cols = ['dec','ra','bin','z']
+        print(f"Loading random catalog columns: {randoms_cols}")
+
+        f = self.open_input('random_cats_source')
+        group = f['randoms']
+
+        cut = self.config['reduce_randoms_size']
+        if 0.0<cut<1.0:
+            N = group['dec'].size
+            sel = np.random.uniform(size=N) < cut
```

Comment:
Not sure I understand what you mean. This is a simple reduction in the number of random objects used in a random catalog. However since we are not using it we could remove it? It was left in for future use, and is a copy of how the random catalogs for lens bins are reduced in the main stage. @joezuntz what do you think?

Reply:
I see now. Fine then!